dcpy/connectors/socrata/publish.py

```diff
@@ -281,39 +275,35 @@ def calculate_pushed_col_metadata(self, our_columns: list[md.DatasetColumn]):
                 }
             )
 
-        new_uploaded_cols = []
-        for uploaded_col in self.uploaded_columns:
+        output_schema = self.get_latest_output_schema()
+
+        for col in self.column_names:
             # Take the Socrata metadata for columns that have been uploaded,
             # modify them to match our metadata.
 
-            # Input columns need to be matched to what's been uploaded,
-            # via the `initial_output_column_id`. Otherwise update
-            # requests are ignored.
-            new_col = copy.deepcopy(uploaded_col)
-
-            our_col = our_cols_by_field_name[uploaded_col["field_name"]]
-            our_col_index = list(our_api_names).index(uploaded_col["field_name"])
+            our_col = our_cols_by_field_name[col]
+            our_col_index = list(our_api_names).index(col)
 
-            new_col["position"] = our_col_index + 1
-            new_col["initial_output_column_id"] = new_col["id"]
+            output_schema.change_column_metadata(col, "position").to(our_col_index + 1)
+            #    .change_column_metadata(col, "initial_output_column_id").to(col.id)
 
-            new_col["is_primary_key"] = (
+            output_schema.change_column_metadata(col, "is_primary_key").to(
                 True if (our_col.checks and our_col.checks.is_primary_key) else False
             )
 
-            new_col["display_name"] = our_col.name
-            new_col["description"] = our_col.description
-            new_uploaded_cols.append(new_col)
+            output_schema.change_column_metadata(col, "display_name").to(our_col.name)
+            output_schema.change_column_metadata(col, "description").to(
+                our_col.description
+            )
+            if our_col.custom and our_col.custom.get("api_name"):
+                logger.info(f"Mapping column {our_col.id} to {col}")
+                output_schema.change_column_transform(col).to(f"`{our_col.id}`")
```

Comment:
this probably should be `our_col.name.lower()` rather than `our_col.id` - as far as I can tell, that's how socrata reads in the datasets